cli/src/cli_util.rs

```diff
@@ -948,6 +949,41 @@ impl WorkspaceCommandHelper {
         }
     }
 
+    fn evaluate_revset_with_modifier_to_single_commit(
+        &self,
+        ui: &mut Ui,
+        revision_arg: &RevisionArg,
+        modifier: Option<RevsetModifier>,
+        expression: &RevsetExpressionEvaluator<'_>,
+        should_hint_about_all_prefix: bool,
+    ) -> Result<Commit, CommandError> {
+        Ok(if modifier == Some(RevsetModifier::Prompt) {
+            let targets: Vec<Commit> = expression.evaluate_to_commits()?.try_collect()?;
```

Comment:
nit: fail on too many commits, as it doesn't make sense on something like `jj new 'prompt:all()'`, as discussed on Discord.

Reply:
and before I forget, this can't be eagerly evaluated as it poses a problem for Google repo.